inc/wpseo-functions.php

```diff
@@ -259,3 +259,56 @@ function wpseo_get_capabilities() {
 	}
 	return WPSEO_Capability_Manager_Factory::get()->get_capabilities();
 }
+
+
+/**
+ * Callback function to pass to the oEmbed's response data that will enable
+ * support for using the image and title set by the WordPress SEO plugin's fields. This
+ * address the concern where some social channels/subscribed use oEmebed data over OpenGraph data
+ * if both are present.
+ *
+ * @param array  $data   - The oEmbed data.
+ * @param object $post   - The current Post object.
+ * @param number $width  - The current post's width (more applicable for attachments).
+ * @param number $height - The current post's height (more applicable for attachments).
+ *
+ * @see https://developer.wordpress.org/reference/hooks/oembed_response_data/ for hook info
+ *
+ * @return array $filter_data - An array of oEmbed data with modified values where appropriate.
+ */
+function wpseo_hook_oembed( $data, $post, $width, $height ) {
+	// Data to be returned.
+	$filter_data = $data;
+
+	// Look for the Yoast meta values (image and title)...
+	$opengraph_title = WPSEO_Meta::get_value( 'opengraph-title', $post->ID );
+	$opengraph_image = WPSEO_Meta::get_value( 'opengraph-image', $post->ID );
+
+	// If yoast has a title set, update oEmbed with Yoast's title.
+	if ( ! empty( $yoast_title ) ) {
+		$filter_data['title'] = $opengraph_title;
+	}
+
+	// If the a Yoast Image was set, update the oEmbed data with the Yoast Image's info.
+	if ( ! empty( $opengraph_image ) ){
+		// Get the image's ID from a URL.
+		$image_id = attachment_url_to_postid( $opengraph_image );
```

Comment:
```suggestion
		$image_id = WPSEO_Image_Utils::get_attachment_by_url( $opengraph_image );
```

Reply:
Good to go, I forgot to commit this in my last push. All set now, thanks!